test/Source/DoctrineORMQuerySourceIteratorTest.php

```diff
@@ -0,0 +1,67 @@
+<?php
+
+/*
+ * This file is part of the Sonata Project package.
+ *
+ * (c) Thomas Rabaix <thomas.rabaix@sonata-project.org>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Exporter\Test\Source;
+
+use Doctrine\ORM\Configuration;
+use Doctrine\ORM\EntityManagerInterface;
+use Doctrine\ORM\Query;
+use Exporter\Test\Source\Fixtures\DoctrineORMQuerySourceIterator;
+use Exporter\Test\Source\Fixtures\ObjectWithToString;
+
+/**
+ * @author Joseph Maarek <josephmaarek@gmail.com>
+ */
+final class DoctrineORMQuerySourceIteratorTest extends \PHPUnit_Framework_TestCase
+{
+    /** @var Query */
+    protected $query;
+
+    protected function setUp()
+    {
+        /** @var EntityManagerInterface|\PHPUnit_Framework_MockObject_MockObject $entityManager */
+        $entityManager = $this->createMock('Doctrine\ORM\EntityManagerInterface');
+        $entityManager->method('getConfiguration')->willReturn(new Configuration());
+        $this->query = new Query($entityManager);
+    }
+
+    /**
+     * @dataProvider getValueProvider
+     */
+    public function testGetValue($value, $expected, $dateFormat = 'r')
+    {
+        $iterator = new DoctrineORMQuerySourceIterator($this->query, array(), $dateFormat);
+        $this->assertSame($expected, $iterator->getValue($value));
+    }
+
+    public function getValueProvider()
+    {
+        $datetime = new \DateTime();
+
+        $data = array(
+            array(array(1, 2, 3), null),
+            array(new \ArrayIterator(array(1, 2, 3)), null),
+            array($datetime, $datetime->format('r')),
+            array($datetime, $datetime->format('Y-m-d H:i:s'), 'Y-m-d H:i:s'),
+            array(123, 123),
+            array('123', '123'),
+            array(new ObjectWithToString('object with to string'), 'object with to string'),
+        );
+
+        if (class_exists('\DateTimeImmutable')) {
```

Comment:
This class will exists always on PHP 7.1 right? Please remove this condition and simplify code

Reply:
Not relevant anymore, 1.x will still support PHP 5.6.

Reply:
So should we keep it or take it off?

Reply:
DateTimeImmutable exists since php 5.5.0 
http://php.net/manual/fr/class.datetimeimmutable.php